sw/device/lib/dif/dif_plic_unittest.cc

```diff
@@ -341,7 +341,7 @@ class IrqCompleteTest : public PlicTest {
 };
 
 TEST_F(IrqCompleteTest, NullArgs) {
-  dif_plic_irq_id_t data;
+  dif_plic_irq_id_t data = 0;
```

Comment:
I am a bit confused about this compile error. Is data expected as a `const *` parameter? If so, is it a bug in the dif?

Reply:
This is either a bug in the DIF or a bug in GCC; doing this merely masks the problem.

(Incidentally, I think we should be running our tests under ASAN and MSAN to avoid running into this in the first place. @cfrantz thoughts?)

Reply:
I think the fix is OK. It looks like `complete_data` is intended to be an input despite being documented as an out-param. We should probably be passing it by value but that can be a separate clean up.

@imphil can you also remove `[out]` from the comment [here](https://github.com/lowRISC/opentitan/blob/57054be3fc2d4d6155783b2383b5f4cddafa23a1/sw/device/lib/dif/dif_plic.h#L290) to avoid future confusion?

Reply:
> (Incidentally, I think we should be running our tests under ASAN and MSAN to avoid running into this in the first place. @cfrantz thoughts?)

I think this would be a generally good thing. I don't know if they would necessarily have caught this though since `data` is never actually accessed (in the tests where `data` is used the function returns 'bad arg' before it accesses anything).